packages/ts/file-router/src/vite-plugin/collectRoutesFromFS.ts

```diff
@@ -74,6 +75,10 @@ export default async function collectRoutesFromFS(
       children.push(await collectRoutesFromFS(new URL(`${d.name}/`, dir), { extensions, logger, parent: dir }));
     } else if (d.isFile() && extensions.includes(extname(d.name))) {
       const file = new URL(d.name, dir);
+      const url = await checkFile(file, logger);
```

Comment:
That's definitely not the best place to check files because there is a lot of files we're skipping. That's why I was checking the file in the array after that.

Reply:
> That's definitely not the best place to check files because there is a lot of files we're skipping. That's why I was checking the file in the array after that.

Could you please elaborate more about the files we are skipping if we do the checking here?
To my understanding, this branch of `else if` is the one adding the entires to the `children` and we only care about those files with the specified extensions, so I couldn't understand what are the skipped files, and why is it important to first add some entires to `children` for empty files (or those that are not exporting a component) and then filter them later. 
A brief description of the rule in my mind: 
Files with `.tsx` or `.jsx` extension that are empty or doesn't have a export default (yet) can be considered non-existent from the FS Router POV. No need to consider their existence in the first place.

Reply:
The idea behind my comment was that we're skipping the files that start with `_`. However, in your implementation, they are also checked. However, each check is IO operation which is expensive. That's why I perform the check only with already added files.

Also, for me, it looks more logically correct when all checks are done in the same place somehow.